File: frontend/src/types/livro.ts
export type Livro = {
  id: string;
  titulo: string | null;
  autor: string | null;
  genero: string | null;
  ano: number | null;
  numero_paginas: number | null;
  descricao: string | null;
  imagem_url?: string | null;
  imagem?: string | null;
  capa_url?: string | null;
  capa?: string | null;
};

export type LivrosPaginados = {
  items: Livro[];
  total: number;
  limite: number;
  offset: number;
};

export type ListarLivrosParams = {
  limite?: number;
  offset?: number;
  busca?: string;
  genero?: string;
};

export type LivroRecomendado = Livro & {
  similaridade: number;
};

export type RecomendacaoPayload = {
  preferencia: string;
  limite: number;
};

export type BookFromAPI = {
  title: string | null;
  author: string | null;
  year: number | null;
  number_of_pages: number | null;
  book_description: string | null;
  category: string | null;
};

export type RecommendationResponse = {
  response: string;
  retrieved_books: BookFromAPI[];
};

File: backend/backend-recomendacao-livros/app/schemas/livro.py
from pydantic import BaseModel, ConfigDict


class LivroResponse(BaseModel):
    id: str
    titulo: str | None
    autor: str | None
    genero: str | None
    ano: int | None
    numero_paginas: int | None
    descricao: str | None
    imagem_url: str | None = None

    model_config = ConfigDict(from_attributes=True)


class LivroListResponse(BaseModel):
    items: list[LivroResponse]
    total: int
    limite: int
    offset: int
